src/components/Calculator.js

```diff
@@ -1,34 +1,73 @@
 import React from 'react';
+import calculate from '../logic/calculate';
 
 class Calculator extends React.Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = {
+      total: null,
+      next: null,
+      operation: null,
+    };
+    this.clickHandler = this.clickHandler.bind(this);
+    this.displayResult = this.displayResult.bind(this);
+  }
+
+  clickHandler(e) {
+    const { total, next } = this.state;
+    if (!total && !next && e.target.classList[0] === 'arithmetic_operand') {
+      return;
+    }
+    this.setState((objName) => calculate(objName, e.target.dataset.buttonName));
+  }
+
+  displayResult() {
+    const { total, next, operation } = this.state;
+    let result = null;
+
+    if (total === null && next === null) {
+      result = 0;
+    } else if (next !== null && total === null) {
+      result = next;
+    } else if (next !== null && total !== null) {
+      if (operation !== null && operation !== undefined) {
+        result = `${total} ${operation} ${next}`;
+      } else {
+        result = total;
+      }
+    } else if (next === null && total !== null) {
+      if (operation !== null && operation !== undefined) {
+        result = `${total} ${operation}`;
+      } else {
+        result = total;
+      }
+    }
+    return result;
   }
```

Comment:
**Optional Changes**
- [ ] DisplayResults looks like it is performing logic. You will want to move it to the `logic` directory in order to separate concerns in your codebase.